<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>901</width>
    <height>613</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Photolab</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,2">
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>-449</y>
         <width>188</width>
         <height>988</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Color</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6" stretch="0,0,0,0,0,0,0,0,0,0,0,0">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QRadioButton" name="rb_HSL">
               <property name="text">
                <string>HSL</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="rb_HSV">
               <property name="text">
                <string>HSV</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Hue</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QSlider" name="hs_Hue">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sb_Hue">
               <property name="maximum">
                <number>360</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Saturation</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QSlider" name="hs_Saturation">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>255</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sb_Saturation">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="l_Light_Value">
             <property name="text">
              <string>Light</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QSlider" name="hs_Light_Value">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>255</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sb_Light_Value">
               <property name="maximum">
                <number>255</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Brightness</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QSlider" name="hs_Brightness">
               <property name="minimum">
                <number>-100</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sp_Brightness">
               <property name="minimum">
                <number>-100</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Contrast</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QSlider" name="hs_Contrast">
               <property name="minimum">
                <number>-100</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="sb_Contrast">
               <property name="minimum">
                <number>-100</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="b_Apply">
             <property name="text">
              <string>Apply</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Filters</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QPushButton" name="b_Negative">
             <property name="text">
              <string>Negative</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_Toning">
             <property name="text">
              <string>Toning</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_Grayscale">
             <property name="text">
              <string>Grayscale</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_Emboss">
             <property name="text">
              <string>Emboss</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_Sharpen">
             <property name="text">
              <string>Sharpen</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_BoxBlur">
             <property name="text">
              <string>Box Blur</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_GaussianBlur">
             <property name="text">
              <string>Gaussian Blur</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_Laplacian">
             <property name="text">
              <string>Laplacian Filter</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_Prewitt">
             <property name="text">
              <string>Prewitt Filter</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_SobelLeft">
             <property name="text">
              <string>Sobel Filter Left</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_SobelRight">
             <property name="text">
              <string>Sobel Filter Right</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_SobelFull">
             <property name="text">
              <string>Sobel Filter Full</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="b_CustomFilter">
             <property name="text">
              <string>Custom Filter</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="CustomLabel" name="labelImage">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>History</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="HistoryWidget" name="sa_History">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>901</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuPhoto">
    <property name="title">
     <string>Photo</string>
    </property>
    <addaction name="actionReverse"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuPhoto"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionReverse">
   <property name="text">
    <string>Reverse</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HistoryWidget</class>
   <extends>QScrollArea</extends>
   <header>HistoryWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CustomLabel</class>
   <extends>QLabel</extends>
   <header>CustomLabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_Brightness</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sp_Brightness</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>348</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>343</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_Contrast</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sb_Contrast</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>408</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_Hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sb_Hue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>148</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_Saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sb_Saturation</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_Light_Value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sb_Light_Value</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sb_Light_Value</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_Light_Value</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sb_Saturation</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_Saturation</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>213</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sb_Hue</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_Hue</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sb_Contrast</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_Contrast</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>408</y>
    </hint>
    <hint type="destinationlabel">
     <x>104</x>
     <y>413</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sp_Brightness</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_Brightness</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>343</y>
    </hint>
    <hint type="destinationlabel">
     <x>104</x>
     <y>348</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
